### PR TITLE
Remove 3.1.4-SNAPSHOT shim code

### PR DESCRIPTION
**Pull request by nartal1**
This fixes https://github.com/NVIDIA/spark-rapids/issues/7409

This PR is to clean up code for spark-314 shim(which isn't released  in Apache Spark).
Updated test which was referencing this shim.

### CODE REVIEW

**Comment by nartal1** [comment]
build

**Comment by nartal1** [comment]
build

**Comment by nartal1** [comment]
> git grep for other occurrences of 314 such as https://github.com/nartal1/spark-rapids/blob/bdc5fe4725a4ada913ffff905f95f7a05cfb9fdc/sql-plugin/src/main/spark311/scala/com/nvidia/spark/rapids/shims/OrcCastingShims.scala#L31

Thank you for catching this. I had done `git grep '{"spark": "314"}' '*.scala' '*.java'` earlier which missed the above occurence.
Removed 314 from OrcCastingShims.scala.
I ran `git grep '314' '*.scala' '*.java' '*.py'` now and don't see any reference to 314. PTAL

**Comment by nartal1** [comment]
build

**Comment by pxLi** [comment]
build

File: integration_tests/src/main/python/arithmetic_ops_test.py [hunk 1]
```diff
@@ -369,7 +369,8 @@ def test_mod_pmod_by_zero(data_gen, overflow_exp):
 def test_cast_neg_to_decimal_err():
     # -12 cannot be represented as decimal(7,7)
     data_gen = _decimal_gen_7_7
-    if is_before_spark_314() or ((not is_before_spark_320()) and is_before_spark_322()):
+    #if is_before_spark_313() or ((not is_before_spark_320()) and is_before_spark_322()):
```
Comment (gerashegalov) [inline integration_tests/src/main/python/arithmetic_ops_test.py:372]: why do we need this comment?

**Review by gerashegalov — DISMISSED** [review]
LGTM

File: integration_tests/src/main/python/arithmetic_ops_test.py (hunk 1 above)
Comment (nartal1) [reply integration_tests/src/main/python/arithmetic_ops_test.py]: My bad. Forgot to remove it after testing.

**Comment by nartal1** [comment]
build

**Comment by nartal1** [comment]
build

**Pull request merged**